internal/config/common.go

```diff
@@ -0,0 +1,84 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/BurntSushi/toml"
+)
+
+func PortToAddress(port int) string {
+	return fmt.Sprintf(":%d", port)
+}
+
+func GenerateAddress(hostname string, port int) string {
+	return fmt.Sprintf("http://%s%s", hostname, PortToAddress(port))
+}
+
+func IsFileEmpty(path string) (bool, error) {
+	fileInfo, err := os.Stat(path)
+
+	if err != nil {
+		return false, err
+	}
+
+	return fileInfo.Size() == 0, nil
+}
+
+func GetConfiguration[T any](filename string, defaultGenerator func() T) (T, error) {
+	exePath, err := os.Executable()
+
+	if err != nil {
+		return defaultGenerator(), err
+	}
+
+	exeDir := filepath.Dir(exePath)
+	configPath := filepath.Join(exeDir, filename)
+
+	fileEmpty, err := IsFileEmpty(configPath)
+
+	if err != nil {
+		return defaultGenerator(), err
+	}
+
+	if fileEmpty {
+		return defaultGenerator(), nil
```

Comment:
These semantics are really weird. Why do we say

- No file -> Error
- Empty file -> default config

Reply:
If we have the semantics of using the default (as opposed to zero) for unset fields, then this is fine, and will happen automaticly (instead of needing this check). I'd be fine with that.